src/event_loop.rs

```diff
@@ -323,56 +313,53 @@ impl<Io> EventLoop<Io>
 
         'write_many: while let Some(mut current) = state.take_current() {
             'write_one: loop {
-                match self.pty.write(current.remaining_bytes()) {
+                match self.pty.writer().write(current.remaining_bytes()) {
                     Ok(0) => {
                         state.set_current(Some(current));
                         break 'write_many;
-                    },
+                    }
                     Ok(n) => {
                         current.advance(n);
                         if current.finished() {
                             state.goto_next();
                             break 'write_one;
                         }
-                    },
+                    }
                     Err(err) => {
                         state.set_current(Some(current));
                         match err.kind() {
-                            ErrorKind::Interrupted |
-                            ErrorKind::WouldBlock => break 'write_many,
+                            ErrorKind::Interrupted | ErrorKind::WouldBlock => break 'write_many,
                             _ => return Err(err),
                         }
                     }
                 }
-
             }
         }
 
         Ok(())
     }
 
-    pub fn spawn(
-        mut self,
-        state: Option<State>
-    ) -> thread::JoinHandle<(EventLoop<Io>, State)> {
+    pub fn spawn(mut self, state: Option<State>) -> thread::JoinHandle<(Self, State)> {
         thread::spawn_named("pty reader", move || {
             let mut state = state.unwrap_or_else(Default::default);
             let mut buf = [0u8; 0x1000];
 
-            let fd = self.pty.as_raw_fd();
-            let fd = EventedFd(&fd);
+            let poll_opts = PollOpt::level() | PollOpt::oneshot();
```

Comment:
There's edge triggering used elsewhere in the code still. Why was this changed to level?

Reply:
There's no good reason, it was just what I changed that happened to make it work. I've tried a number of times to figure it out but haven't been able to.

Basically something about the change (5b444375 -> windows-linux) meant that we don't read all of the data from the TTY Read pipe/input channel and block on the next `poll.poll` call.

Reply:
Hmm, no idea what that could have been. Before we merge this, I want to try the benchmarks with edge triggering on linux again just to see if it has an impact. There's nothing else in this PR that seems like it should have any effect on performance.

Reply:
Fixed in 5dcd5cd. It was actually an issue with not reregistering the input channel which then didn't recieve any more events because of `PollOpt::Oneshot`.

Haven't tested performance, my guess is it won't change.

Reply:
```shell
❯ hyperfine "./alacritty_master -e '/usr/bin/cat' ./scroll_test.vte" "./alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte"
Benchmark #1: ./alacritty_master -e '/usr/bin/cat' ./scroll_test.vte
  Time (mean ± σ):      7.811 s ±  0.157 s    [User: 2.727 s, System: 11.529 s]
  Range (min … max):    7.647 s …  8.090 s
Benchmark #2: ./alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte
  Time (mean ± σ):      7.915 s ±  0.189 s    [User: 2.761 s, System: 11.609 s]
  Range (min … max):    7.660 s …  8.274 s
Summary
  './alacritty_master -e '/usr/bin/cat' ./scroll_test.vte' ran
    1.01 ± 0.03 times faster than './alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte'
```

Reply:
From this I'm going to say the performance is equal. 

I even bumped the number of runs from 10 to 20 and the difference range increased (was `± 0.03`).

```shell
❯ hyperfine "./alacritty_master -e '/usr/bin/cat' ./scroll_test.vte" "./alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte" --min-r
Benchmark #1: ./alacritty_master -e '/usr/bin/cat' ./scroll_test.vte
  Time (mean ± σ):      7.921 s ±  0.264 s    [User: 2.791 s, System: 11.651 s]
  Range (min … max):    7.585 s …  8.532 s
Benchmark #2: ./alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte
  Time (mean ± σ):      7.842 s ±  0.208 s    [User: 2.706 s, System: 11.577 s]
  Range (min … max):    7.606 s …  8.396 s
Summary
  './alacritty_windows -e '/usr/bin/cat' ./scroll_test.vte' ran
    1.01 ± 0.04 times faster than './alacritty_master -e '/usr/bin/cat' ./scroll_test.vte'

```